### PR TITLE
Global type variable names in error messages

### PR DESCRIPTION
**Pull request by jstolarek**
See #43. Don't merge yet.

### CODE REVIEW

File: tests/shredding/config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+database_driver=postgresql
```
Comment (fehrenbach) [inline tests/shredding/config]: You accidentally added the shredding config.

File: types.ml [hunk 2]
```diff
@@ -1749,7 +1753,7 @@ struct
                                           && is_present (FieldEnv.find v fields))
                         true values in
 
-       let arrow =
+       let arrow () =
```
Comment (jstolarek) [inline types.ml]: One of them but unfortunately there's still more :-/

**Comment by jamescheney** [comment]
Is this now ready to be merged?

**Comment by jstolarek** [comment]
> Is this now ready to be merged?

There is one more thing that remains to be done: always displaying type variable and effect names when printing error messages. This should be simple to do but I need to consult @slindley about one implementation detail.

Alternatively, we can merge this in - because it essentially does what it should - and then merge the mentioned change separately.

**Comment by jstolarek** [comment]
Merged as 3e19badbb65df83a9fd15bdfbffb85334745719b

**Pull request closed without merging**